### PR TITLE
Simplified contact form

### PR DESCRIPTION
**Pull request by JevgenijVisockij**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies contact form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Simplifying Shop Parameters -> Contact add/edit form. Everything must look the same.

:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType by ContactType. This means if any module extends this type they will get an exception upon upgrading to PS version containing changes in this PR.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21243)

### CODE REVIEW

File: classes/Contact.php
Comment (JevgenijVisockij) [inline classes/Contact.php:73]: Right now constraints of the form don't make sense. There is validation isCleanHtml here and in formType but because it's type is TYPE_STRING here it means it won't even save simple HTML like `<div>`. I don't know if modifying old object is "allowed" in this case.

If it's not then I will change constraints to reflect that it won't actually save HTML.

Comment (matks) [reply classes/Contact.php:73]: Hi @JevgenijVisockij it seems you have found an inconsistency!

Modifying old object is allowed when we find an inconsistency, it's even a good thing! The tricky part is to find out
- is the type right (so it's a string) and the constraints wrong?
or
- is the type wrong (so it's a HTML field) and the constraints right?

I tested a bit and I think that this field is actually indeed a string. For me the constraints are wrong 🤔  . Because
- when I used current form to save HTML the markup is removed to transform it into plain string
- the form input is not a HTML WYSIWYG and I dont expect merchants to write hardcoded HTML

**Review by matks — COMMENTED** [review]
I think we should keep type string and modify the constraints instead. The code is valid for me, excluding this item 👍

File: classes/Contact.php
Comment (Progi1984) [reply classes/Contact.php:73]: Changes needed

Comment (JevgenijVisockij) [reply classes/Contact.php:73]: > "when I used current form to save HTML the markup is removed to transform it into plain string" 

This happens exactly because of `'type' => self::TYPE_STRING,` so it was a question whether the object model was right or form is right. 

> the form input is not a HTML WYSIWYG and I dont expect merchants to write hardcoded HTML

I agree here, I am gonna edit the constraints so that it won't actually allow you to input HTML.

**Comment by JevgenijVisockij** [comment]
Blocked by https://github.com/PrestaShop/PrestaShop/pull/20243/ because it also deals with fields where HTML tags should throw error. Once that is approved/merged same can be used here.

File: src/Core/ConstraintValidator/Constraints/NoTags.php
Comment (JevgenijVisockij) [inline src/Core/ConstraintValidator/Constraints/NoTags.php:1]: This used to be in https://github.com/PrestaShop/PrestaShop/pull/20243/files PR but it's no longer needed there and needed here so I moved it.

File: src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]:
```suggestion
                                'The "%s" field is invalid. HTML tags are not allowed.',
```
If I am not mistaken, it will look like 'The description field is invalid. HTML tags are not allowed.'

Comment (Julievrz) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]: Do we need the quotation marks here?

**Review by Julievrz — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php
Comment (JevgenijVisockij) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]: ```The "description" field is invalid. HTML tags are not allowed.``` so yeah with quotations. Regarding if they are needed I am not sure. As far as if they are needed I am not sure, I see benefit of name being in quotations, but they also add extra noise.

**Review by Progi1984 — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php
Comment (Julievrz) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]: Mhm, I'd say we keep it. for now To stay consistent with other pages where we use this string, like in invoices form.

Comment (Julievrz) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]: It's difficult to answer that question because the behavior is different everywhere on the software right now. :( Either we keep it to stay consistent with this PR (https://github.com/PrestaShop/PrestaShop/pull/20243/files) or we modify it on both. I agree with you that they should be there (if we follow grammatical rules) but you're right to say that they are adding extra noise...

Comment (Julievrz) [reply src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]: @LouiseBonnard What would you do? 😉

File: src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php
Comment (Julievrz) [inline src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/Contact/ContactType.php]:
```suggestion
                                'The "%s" field is invalid. HTML tags are not allowed.',
```
So I think we should keep it to stay consistent with other PRs.

**Review by Progi1984 — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by Julievrz — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

I found some issues:

**First issue**

1. Disable debug mode
2. Go to BO > Shop Parameters > Contact page 
3. Add new contact
4. In the Title field add an expression > 265 characters 
5. Save 
6. See error 
image


**Second issue**

1. Go to BO > Shop Parameters > Contact page 
2. Add new contact
3. Click on help => redirection to another page => it should be opened in modal

image


Thank you!

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by JevgenijVisockij** [comment]
> Hi @JevgenijVisockij,
> 
> I found some issues:
> 
> **First issue**
> 
>     1. Disable debug mode
> 
>     2. Go to BO > Shop Parameters > Contact page
> 
>     3. Add new contact
> 
>     4. In the Title field add an expression > 265 characters
> 
>     5. Save
> 
>     6. See error
>        image
> 
> 
> **Second issue**
> 
>     1. Go to BO > Shop Parameters > Contact page
> 
>     2. Add new contact
> 
>     3. Click on help => redirection to another page => it should be opened in modal
> 
> 
> image
> 
> Thank you!

Thanks for Q&A. Fixed both issues.

**Review by atomiix — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1YTN-hzd_6HSgOXKpZ0z6a0ldG6Q1KisB/view?usp=sharing

Thank you!

**Comment by Progi1984** [comment]
Thanks @JevgenijVisockij & @khouloudbelguith

**Pull request merged**
